pkg/api/vlabs/validate.go

```diff
@@ -300,6 +300,10 @@ func (a *Properties) ValidateOrchestratorProfile(isUpdate bool) error {
 			}
 
 			if a.IsAzureStackCloud() {
+				if common.IsKubernetesVersionGe(a.OrchestratorProfile.OrchestratorVersion, "1.21.0") && !to.Bool(o.KubernetesConfig.UseCloudControllerManager) {
```

Comment:
This validation happens *before* defaults enforcement, so this is going to throw an error every time a user tries to create a cluster with an empty `UseCloudControllerManager` configuration